test/standalone/test-debuglet.js

```diff
@@ -280,5 +280,15 @@ describe(__filename, function(){
       assert.deepEqual(Debuglet.mapSubtract({}, b), []);
     });
   });
+
+  describe('format', function() {
+    it('should be correct', function() {
+      assert.deepEqual(Debuglet.format('hi', [5]), 'hi');
+      assert.deepEqual(Debuglet.format('hi $0', [5]), 'hi 5');
+      assert.deepEqual(Debuglet.format('hi $0 $1', [5, 'there']), 'hi 5 there');
+      assert.deepEqual(Debuglet.format('hi $0 $1', [5]), 'hi 5 $1');
+      assert.deepEqual(Debuglet.format('hi $0 $1 $0', [5]), 'hi 5 $1 5');
```

Comment:
Some more tests to add: "$$", "$$0", "$00", "hi$01". Also

'hi $0', ['$1', 'foo'] should result in 'hi $1'.